### PR TITLE
[SDK & CLI]Support provider configurable in some command

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Support provider configurable in below scenarios:
1. PFClient
```
PFClient(config={"connection.provider": "azureml"})
```
2. flow test
```
pf flow test --flow . --config connection.provider=azureml
```
3. flow serve
```
pf flow serve --source . --config connection.provider=azureml
```

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    finally:
+        sys.argv = origin_argv
+        os.chdir(origin_cwd)
+
+
+@pytest.mark.cli_test
+@pytest.mark.e2etest
+class TestCli:
+    # PF cli test is here because when designate connection provider to remote, we need azure dependencies.
+    def test_pf_flow_test(self, remote_workspace_resource_id):
```
Comment (lalala123123) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_cli.py:38]: Could you add a test case about chat flow
You can refer to this test case https://github.com/microsoft/promptflow/blob/6260a318fe46ebb1ed84ca9127c6777e66844362/src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py#L766C1-L766C1

File: src/promptflow/tests/sdk_cli_test/unittests/test_pf_client.py [hunk 2]
```diff
@@ -49,6 +49,16 @@ def test_connection_provider(self):
...
             assert isinstance(client.connections, LocalAzureConnectionOperations)
 
+        with mock.patch(target) as mocked:
+            mocked.get_instance.return_value.get_connection_provider.return_value = (
+                "azureml:"
+                + RESOURCE_ID_FORMAT.format(
+                    "96aede12-2f73-41cb-b983-6d11a904839b", "promptflow", AZUREML_RESOURCE_PROVIDER, "promptflow-eastus"
+                )
+            )
+            client = PFClient(connection_provider="local")
```
Comment (wangchao1230) [inline src/promptflow/tests/sdk_cli_test/unittests/test_pf_client.py:59]: have we considered adding a more general way to specify config like below:
PFClient(config = {"connection.provider": "local"}

File: src/promptflow/promptflow/_cli/_params.py [hunk 3]
```diff
@@ -306,4 +306,12 @@ def add_param_verbose(parser):
     )
 
 
+def add_param_connection_provider(parser):
+    parser.add_argument(
+        "--connection-provider",
+        type=str,
+        help=argparse.SUPPRESS,
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_params.py:313]: do we expect end user to type this in their command?  if yes, this should not be suppressed

Comment (brynn-code) [reply src/promptflow/promptflow/_cli/_params.py:314]: No, just for ai studio for now.

**Review by wangchao1230 — APPROVED** [review]

**Review by lalala123123 — APPROVED** [review]

**Pull request merged**